rsconf/component/devbox.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+"""Sirepo development in a docker container on a remote machine
+
+:copyright: Copyright (c) 2019 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern import pkio
+from pykern import pkjson
+from pykern.pkcollections import PKDict
+from pykern.pkdebug import pkdp
+from rsconf import component
+import subprocess
+
+_PASSWD_SECRET_JSON_F = "devbox_auth.json"
+
+
+class T(component.T):
+    def install_resource(self, *args, **kwargs):
+        super().install_resource(*args, module_name="devbox", **kwargs)
+
+    def internal_build_compile(self):
+        from rsconf import systemd
+
+        if "user_name" not in self:
+            for u in self.hdb.devbox.users.keys():
+                t = T(u, self.buildt, user_name=u)
+                self.buildt.build_component(t)
+            return
+        self.buildt.require_component("docker", "network")
+        jc, _ = self.j2_ctx_init()
+        z = jc.devbox
+        z.setdefault("volumes", ["jupyter", "src"])
+        z.host_d = systemd.unit_run_d(jc, "devbox_" + self.user_name)
```

Comment:
seems like compt.name could be devbox_user_name and set that on create. Then everything will be consistent.

Reply:
fixing the compt.name removed the need for supplying run_d 5d2406d

Reply:
Whoops, previous comment was on the wrong comment of yours. 

Thanks for seeing compt.name. Cleaned things up 5d2406d